code/modules/mob/living/carbon/carbon.dm

```diff
@@ -81,23 +81,28 @@
 	var/hurt = TRUE
 	var/extra_speed = 0
 	if(throwingdatum.thrower != src)
-		extra_speed = min(max(0, throwingdatum.speed - initial(throw_speed)), 3)
+		extra_speed = min(max(0, throwingdatum.speed - initial(throw_speed)), CARBON_MAX_IMPACT_SPEED_BONUS)
 
-	if(istype(throwingdatum, /datum/thrownthing))
+	if(istype(throwingdatum))
 		hurt = !throwingdatum.gentle
-	if(hit_atom.density && isturf(hit_atom))
-		if(hurt)
-			Paralyze(20)
+	if(hurt && hit_atom.density)
+		if(isturf(hit_atom))
+			Paralyze(2 SECONDS)
 			take_bodypart_damage(10 + 5 * extra_speed, check_armor = TRUE, wound_bonus = extra_speed * 5)
+		else if(isstructure(hit_atom) && extra_speed)
```

Comment:
does this affect throwing people into vending machines?

Reply:
also, why do the structure and not-carbon checks check for extra_speed, but not the turf check? shouldn't the former be changed to match the standards of the latter?

Reply:
Walls and turfs are harder surfaces, so any impact with them is bad, whereas windows you can bounce off. Also vending machines check tipping in their hitby() so they won't be affected